addons/rating/models/mail_thread.py

```diff
@@ -204,3 +205,17 @@ def _message_update_content(self, message, body, *args, rating_value=None, **kwa
         return super()._message_update_content(
             message, body, *args, rating_value=rating_value, **kwargs
         )
+
+    def _thread_to_store(self, store: Store, fields, *, request_list=None):
+        super()._thread_to_store(store, fields, request_list=request_list)
+        for thread in self:
+            if (
+                request_list
+                and "ratingStats" in request_list
+                and hasattr(thread, "rating_get_stats")
+            ):
+                store.add(
+                    # sudo: rating.rating - access to rating stats on accessible thread is allowed
+                    # see _process_request_for_all
+                    thread, {"rating_stats": thread.sudo().rating_get_stats()}, as_thread=True
+                )
```

Comment:
I'm trying to avoid having store.add inside loops, and work on recordset instead.
so have `store.add(self, ...)` where you define a list of fields. Thread-specific value can be computed with Attr(name, fn) as field item, or having field as string and defining its `_field_store_repr` if the compute is to be used in several places.